### ISSUE TITLE
Rework custom scoring functionality

### ISSUE THREAD

**Issue by mglowacki100** [rhiever/tpot#156]
Now, as far I understood there is only "balanced class accuracy" as default;  precision, recall and f1 are hard-coded in: http://rhiever.github.io/tpot/examples/Custom_Scoring_Functions/
Do you consider to add other metrics either as hard-coded custom scoring function or wrappers on:
http://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics ? I'm especially interested in log_loss metric.

**Comment by rhiever** [comment]
As of now, you would have to write a wrapper function as shown in the docs page you listed. Now that I look at that, though, that doesn't seem very ideal. I'm going to turn this issue into an enhancement request to improve the custom scoring functionality. It should be possible to pass an arbitrary scoring function from sklearn and allow TPOT to use that, without writing a custom wrapper function.

**Comment by bartleyn** [comment]
I can whip something up for this. I think there was an initial concern about how to optimize the scoring functions for DataFrames, but it should be pretty straightforward to just ship off all those concerns to the underlying sklearn functions.

**Comment by rhiever** [comment]
Thanks @bartleyn!

**Comment by mglowacki100** [comment]
Great !!!

**Comment by rhiever** [comment]
Just wanted to ping you, @bartleyn, to see how this was coming along.

**Comment by mglowacki100** [comment]
@bartleyn when do you plan to release this functionality?

**Comment by bartleyn** [comment]
Sorry, I've been horribly negligent of my duties on this (I had a personal emergency to deal with). I've got the lion's share of it done, but need to test edge cases (multiclass, multilabel, etc) a little more thoroughly before putting up a PR. I should comfortably have something by the weekend.

**Comment by rhiever** [comment]
This change will be in the upcoming 0.5 release.
